tests/test_agents.py

```diff
@@ -749,6 +750,46 @@ def test_answers_are_striped() -> None:
     response.model_dump_json()
 
 
+@pytest.mark.asyncio
+async def test_sequential_tool_calls(agent_test_settings: Settings):
+
+    SLEEP_TIME = 2.0
+
+    async def fake_gather_evidence(*args, **kwargs) -> str:  # noqa: ARG001
+        await asyncio.sleep(SLEEP_TIME)
+        return "fake evidence"
+
+    question = "How can you use XAI for chemical property prediction?"
+    env = PaperQAEnvironment(
+        query=QueryRequest(query=question, settings=agent_test_settings),
+        docs=Docs(),
+    )
+    await env.reset()
+
+    gather_tool = next(
+        tool for tool in env.tools if tool.info.name == GatherEvidence.TOOL_FN_NAME
+    )
+
+    with patch.object(gather_tool, "_tool_fn", fake_gather_evidence):
+        tic = time.time()
+        await env.step(
+            ToolRequestMessage(
+                tool_calls=[
+                    ToolCall.from_name(
+                        "gather_evidence",
+                        question="XAI for chemical property prediction",
+                    ),
+                    ToolCall.from_name(
+                        "gather_evidence",
+                        question="XAI for chemical property prediction",
+                    ),
+                ]
+            )
+        )
+
+        assert time.time() - tic > 2 * SLEEP_TIME  # since they are sequential
```

Comment:
Nice PR and happy to see a test change. 

This test directly mirrors the updated implementation (`ordered=True`) not so much the actual issue (concurrent `gather_evidence` clobbering `state.session.question`). This pattern means this new test isn't useful if we want to fix the race condition(s) in our tools and remove `ordered=True`.

It might be better to move the `gather_evidence` stub here to manipulate `state.session.question` 2+ times to emulate the race condition being faced.

Also, would be good to use `GradablePaperQAEnvironment` and then move this to the below test class `TestGradablePaperQAEnvironment`